renovate.json5

```diff
@@ -3,9 +3,6 @@
   extends: [
     'local>openameba/renovate-config:main.json5',
     'local>openameba/renovate-config:npm.json5',
-    ':automergeAll'
-  ],
-  lockFileMaintenance: {
-    automerge: true
-  }
+    'local>openameba/renovate-config:automerge-all.json5'
```

Comment:
q. main.json5にもlockFileMaintenance:enableの記載がありますが、動作的にはextendsの後勝ち(この場合は自動マージ)になりますか？

Reply:
ふーむ、確かに、把握しておいた方がいいなと思いextendsした後の設定をrenovateコマンドから出力できるのではないかといろいろ調べてみていたんですが、手元では簡単に出力できなかったので実際にどうなるのかはもうちょっと調べてみます。

おそらくextendsを配列で指定する以上、後勝ちではあるはずで、ただオブジェクトのマージがdeepなのかshallowなのかは把握してないのでautomerge系にはlockFileMaintenanceについても書いた、という感じです。後に書くはず、という前提の基で。

Reply:
How to get resolved config:

```bash
RENOVATE_TOKEN=xxx LOG_LEVEL=info npx renovate --platform=github --dry-run --print-config=true openameba/renovate-config | tee log.txt
```

or

```bash
LOG_LEVEL=info npx renovate --token=xxx --platform=github --dry-run --print-config=true openameba/renovate-config | tee log.txt
```

`RENOVATE_TOKEN` is your Personal Access Token, needs `repo` scope

Reply:
☝️これリモートリポジトリのファイルしかチェックできんな

- https://medium.com/@asimmon/locally-test-and-validate-your-renovate-configuration-files-3cd4429b63e2
- https://dev.to/corrupt952/debugging-renovate-locally-4k8

---

> When an array or object configuration option is `mergeable`, it means that values inside it will be added to any existing object or array that existed with the same name.

via: https://docs.renovatebot.com/configuration-options/#configuration-options

とのことで

> mergeable: true

via: https://docs.renovatebot.com/configuration-options/#lockfilemaintenance

なのでlockFileMaintenanceに関しては後勝ちで値はdeep-mergeだそうで配下の値も個別に後勝ちということになりますかね。